targets/TARGET_NORDIC/TARGET_NRF5/sleep.c

```diff
@@ -18,15 +18,21 @@
 #include "mbed_interface.h"
 #include "softdevice_handler.h"
 #include "nrf_soc.h"
+#include "nrf_timer.h"
 
 // Mask of reserved bits of the register ICSR in the System Control Block peripheral
 // In this case, bits which are equal to 0 are the bits reserved in this register
 #define SCB_ICSR_RESERVED_BITS_MASK     0x9E43F03F
 
 #define FPU_EXCEPTION_MASK 0x0000009F
 
-void hal_sleep(void)
+static void sleep_common(bool deepsleep)
```

Comment:
I'm not sure that it's necessary to factor things in sleep_common. The conditional bits introduced when `deepsleep` is true are a prolog and epilog of the `hal_sleep` function. Why not have those in `hal_deepsleep`  and remove the conditions ?

Reply:
So you suggest to move `sleep_common` code to `hal_sleep` and implement `hal_deepsleep` as follows:
```
- disable usticker
- call hall_sleep()
- enable usticker
```

I think it is a good fix which will increase code readability.

Reply:
Fixed.